variables.tf

```diff
@@ -102,8 +102,4 @@ variable "exclusions" {
   default = []
 }
 
-variable "trusted_client_certificate_data" {
-  type        = string
-  default     = null
-  description = "This variable will allow to send trusted client certificate data as string"
-}
+variable "trusted_client_certificate_data" {}
```

Comment:
Is this used/needed now?

Reply:
The trusted_client_certificate_data variable is still used.

The trusted_client_certificate_data variable is set by the code that calls this module to a map containing client certificate names and the certificate themselves.

The dynamic trusted_client_certificate block in main.tf takes the client certificate name specified in the rootca_certificate_name property and looks it up in the trusted_client_certificate_data map to get the certificate, which it then uses to set the data property of the block.

The code in main.tf that looks up the certificate uses a conditional expression.  This [returns false](https://github.com/hmcts/terraform-module-apim-application-gateway/blob/31a7a9ebce779f41ad597d3fc3f8b0b665f9776f/main.tf#L231) if no certificate exists with that name.  This probably isn't correct - perhaps it should return an empty string or a default certificate?